packages/devtools_app/lib/src/provider/provider_screen.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:provider/provider.dart' as provider show Provider;
+
+import '../banner_messages.dart';
+import '../screen.dart';
+import '../split.dart';
+import './instance_viewer/instance_details.dart';
+import './instance_viewer/instance_providers.dart';
+import './instance_viewer/instance_viewer.dart';
+import 'provider_list.dart';
+
+final _hasErrorProvider = Provider.autoDispose<bool>((ref) {
+  if (ref.watch(providerIdsProvider) is AsyncError) return true;
+
+  final selectedProviderId = ref.watch(selectedProviderIdProvider).state;
+
+  if (selectedProviderId == null) return false;
+
+  final instance = ref.watch(
+    rawInstanceProvider(InstancePath.fromProviderId(selectedProviderId)),
+  );
+
+  return instance is AsyncError;
+});
+
+class ProviderScreen extends Screen {
+  const ProviderScreen()
+      : super.conditional(
+          id: id,
+          requiresLibrary: 'package:provider/',
+          title: 'Provider',
+          requiresDebugBuild: true,
+          icon: Icons.palette,
+        );
+
+  static const id = 'provider';
+
+  @override
+  Widget build(BuildContext context) {
+    return const ProviderScreenBody();
+  }
+}
+
+class ProviderScreenBody extends ConsumerWidget {
+  const ProviderScreenBody({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, ScopedReader watch) {
+    final splitAxis = Split.axisFor(context, 0.85);
+
+    // A provider will automatically be selected as soon as one is detected
+    final selectedProviderId = watch(selectedProviderIdProvider).state;
+
+    return ProviderListener<bool>(
+      provider: _hasErrorProvider,
+      onChange: (context, hasError) {
+        if (hasError) showProviderErrorBanner(context);
+      },
+      child: Split(
+        axis: splitAxis,
+        initialFractions: const [0.33, 0.67],
+        children: [
+          const _SplitBorder(child: ProviderList()),
+          Column(
+            children: [
+              if (selectedProviderId != null) ...[
+                Expanded(
+                  child: _SplitBorder(
+                    child: InstanceViewer(
+                      rootPath: InstancePath.fromProviderId(selectedProviderId),
+                    ),
+                  ),
+                ),
+              ] else
+                const Expanded(
+                  child: _SplitBorder(child: SizedBox.expand()),
```

Comment:
Then if there is not a provider selected, the title should be '[No provider selected]' or similar. This will match what we do in the performance page:

![Screen Shot 2021-04-14 at 12 58 22 PM](https://user-images.githubusercontent.com/43759233/114774493-195fed80-9d25-11eb-80c6-1d0b47798d76.png)
![Screen Shot 2021-04-14 at 1 01 28 PM](https://user-images.githubusercontent.com/43759233/114774509-1ebd3800-9d25-11eb-830c-9ec53d87cc82.png)